lib/haskell/natural4/test/LS/TextuaL4Spec.hs

```diff
@@ -0,0 +1,256 @@
+    "start date"   IS A	Date ;
+    ended   IS A	Boolean
+§
+DECLARE	Claim
+HAS	date IS A	Date ;
+    outcome IS	ONE OF	Successful, Unsuccessful ;
+    amount IS A	Money ;
+    "general exclusions apply" IS A	Boolean
+§
+DECLARE	Accident
+HAS	injuries		IS	SET OF	loss ;
+   "general exclusions apply" IS A	Boolean ;
+   "juvenile limit applies" IS A	Boolean ;
+   "triple benefits apply" IS A	Boolean ;
+   "double benefits apply" IS A	Boolean ;
+   "risk percentage" IS A	Number
+§
+DECLARE	Benefit
+HAS	"initial sum assured"			IS A	Number ;
+    stepUpSumAssured			IS A	Number ;
+    name			IS A	String ;
+    claim			IS A	Claim
+§
+DECLARE	PlanAF			IS	ONE OF	planA, planB, planC, planD, planE, planF
+§
+DECLARE	Plan14			IS	ONE OF	plan1, plan2, plan3, plan4
+|]
+
+pauRules = intercalate "§" [pau0, pau4, toplevel, subsidiary]
+pau0 = [r|
+§ PAU0
+GIVEN	addBenefit IS A Number ;
+      otherBenefits IS LIST OF	Number ;
+      policy IS A Policy ;
+      policyHolder IS A PolicyHolder ;
+      accident IS AN Accident ;
+      illness IS A Claim ;
+      "user input" IS A		Dictionary
+GIVETH	"The Answer" IS A Number
+DECIDE	"The Answer" IS	"accident branch"   IF "user input's"	accident_claim IS	selected ;
+        "The Answer" IS "act of god branch"	IF	"user input's"	"act of god" IS	selected ;
+        "The Answer" IS "illness branch"		OTHERWISE
+
+WHERE	"accident branch"		IS	excludedZero  IF "ADD is disqualified entirely" ;
+      "accident branch" IS "ADD benefit"    OTHERWISE ;
+
+      "illness branch"  IS	excludedZero    IF		illness	IS	disqualified ;
+      "illness branch" IS policy's benMR    OTHERWISE ;
+
+	    "ADD is disqualified entirely"
+                      IF ANY ( policyHolder's age	>=	75
+                            , accident's "general exclusions apply"
+                            , policy's ended ) ;
+	    excludedZero		IS	0 ;
+	    "ADD benefit"		IS MIN ( SUM ( addBenefit
+                                   , otherBenefits)
+                             , "risk cap"
+                             ) ;
+
+	    illness	IS	disqualified
+                      IF	ANY ( illness's "general exclusions apply"
+                              , policy's ended )
+|]
+
+pau4 = [r|
+§ PAU4
+GIVEN	policy IS A		Policy ;
+      policyHolder IS A		PolicyHolder ;
+      accident IS AN		Accident
+GIVETH	"Step 4" IS A		Number
+DECIDE	"Step 1" IS	"claimable limited base ADD benefit"			IF	"there were past ADD payouts" ;
+        "Step 1" IS "base ADD benefit"			OTHERWISE ;
+        "Step 2" IS	"juvenile limited"			IF	accident's 	"juvenile limit applies" ;
+        "Step 2" IS "Step 1"			OTHERWISE ;
+        "Step 3" IS	"multiplied by double triple benefit" ;
+        "Step 4" IS	"Step 3" "discounted by"	accident's "risk percentage"
+
+WHERE	"base ADD benefit" IS	policy's benADD ;
+	    "there were past ADD payouts" IF	policyHolder's "past ADD payouts" >	0 ;
+	    "claimable limited base ADD benefit" IS	"claimable limit"	"-" policyHolder's "past ADD payouts" ;
```

Comment:
If you want to use arbitrary symbols as text, you need to put them in strings, like with "-". Other way to do the same line would be this:
```
MINUS ( "claimable limit", policyHolder's "past ADD payouts" )
```